src/tools/perf/lib/ucp_tests.cc

```diff
@@ -275,11 +274,16 @@ class ucp_perf_test_runner {
                               const ucp_am_recv_param_t *rx_params)
     {
         ucs_assert(!(rx_params->recv_attr & UCP_AM_RECV_ATTR_FLAG_DATA));
-        ucs_assert(length == ucx_perf_get_message_size(&m_perf.params));
+        /* Data length can be either 1 (on sender side when receive a final ack)
+         * or expected payload size on the receiver side. */
+        ucs_assertv((length == 1) ||
```

Comment:
also we can change it like this 
```suggestion
        ucs_assertv((length == 1) && (rte_call(&m_perf, group_index) == 0) ||
```

Reply:
IMO we cannot avoid rte_call for every data packet to determine that we are the server and the payload size should be >1. unless we cache it in the class or provide different AM callback.

Reply:
it can be inside assert as i mentioned in this comment

Reply:
yes

Reply:
If we go this way, then I would just cache this index inside a member var

Reply:
Finally I implemented it in a slightly different manner: I adjust `m_am_rx_length` on the sender side, and use this existing variable for assertions. Btw, there was one more place where similar assertion was failing, so this change fixes both of them